### PR TITLE
Automate building of SBO bundle and index images usable in CatalogSources

### PR DESCRIPTION
**Pull request by pmacik**
### Motivation

Currently there is no direct way to produce SBO bundle and index images.

Jira issue: [APPSVC-766](https://issues.redhat.com/browse/APPSVC-766)

### Changes

This PR:
* Adds Makefile targets for building such operator bundle index image that can be used in `CatalogSources`.
* Adds `release-operator-bundle` Makefile target to run all the necessary steps to do a release to quay such as building operator, operator bundle and bundle index, and pushing it to quay.
* Adds that release target into Travis job that builds operator bundle for each merge-to-master, so that an index image is build for each merge to master, too.

### Testing

1. Set the following env variables:
* `QUAY_TOKEN`
* `QUAY_USERNAME`
* `OPERATOR_GROUP` ... usually same as Quay username
2. Run `make release-operator-bundle`
3. Create a catalog source referencing the above index image
```sh
kubectl apply -f - << EOD
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: sbo-test-catalog
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: REPLACE_IMAGE
  displayName: Community Operators
  publisher: Red Hat
EOD
```
4. Install SBO via OperatorHub or Subscription YAML as usual with the appropriate CSV version

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,11 +38,19 @@ jobs:
...
+        - mv opm $GOPATH/bin/
+
       script:
         # Run merge-to-master-release
         - make merge-to-master-release
         - make push-to-manifest-repo
         - make prepare-bundle-to-quay
+        - make build-bundle-image
+        - make push-bundle-image
+        - make build-bundle-index-image
```
Comment (pedjak) [inline .travis.yml:53]: since we are adding more Makefile rules, instead let's try to consolidate and have just one rule that does what we want, something like `push-bundle-index-image` - the rest of the targets should be dependencies to it.

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 channels:
-- currentCSV: service-binding-operator.vBUNDLE_VERSION
-  name: beta
 - currentCSV: service-binding-operator.v0.1.1-364
   name: alpha
+- currentCSV: service-binding-operator.vBUNDLE_VERSION
+  name: beta
```
Comment (pedjak) [inline deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml:5]: this is no-op change, please revert.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml (hunk 2 above)
Comment (pmacik) [reply deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml]: Right, thx

File: .travis.yml [hunk 3]
```diff
@@ -37,11 +38,13 @@ jobs:
         - chmod +x operator-sdk
         - mv operator-sdk $GOPATH/bin/
 
+        # Download OPM tool
```
Comment (shvasude) [inline .travis.yml:41]: what is OPM tool? any references?

Comment (pmacik) [reply .travis.yml:41]: Taken from https://github.com/operator-framework/operator-registry: `opm generates and updates registry databases as well as the index images that encapsulate them.`

It's THE tool used to build the index image in `build-bundle-index-image` target

File: .gitignore [hunk 4]
```diff
@@ -2,6 +2,7 @@
 build/_output
 build/_test
 build/bin
+bundle_tmp*
```
Comment (pedjak) [inline .gitignore:5]: let's create bundle under `build/bundle` folder.

File: .travis.yml [hunk 5]
```diff
@@ -83,5 +83,4 @@ jobs:
         - kubectl cluster-info
         # install OLM
         - curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v${OLM_VERSION}/install.sh | bash -s v${OLM_VERSION}
-      script: make test-acceptance
-
+      script: travis_wait 60 make test-acceptance
```
Comment (pedjak) [inline .travis.yml:86]: let's do this in a separate PR.

File: .travis.yml [hunk 6]
```diff
@@ -31,7 +31,7 @@ jobs:
   include:
     - stage: Push bundle and index images to Quay ()
       #if: branch = master AND type = push
-      if: pull_request
+      if: type = pull_request
```
Comment (pedjak) [inline .travis.yml:34]: why this change?

Comment (pmacik) [reply .travis.yml:34]: Originally, the job is for "on push-to-master" and the for PR is for testing purposes... before this is merged, I'll change it back to push-to-master

Comment (pmacik) [reply .travis.yml:34]: and this particuar commit is fixing a typo (missing `type = ` in the condition)

**Comment by pmacik** [comment]
Closing in favour of https://github.com/redhat-developer/service-binding-operator/pull/770
